src/App.tsx

```diff
@@ -1,26 +1,118 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+// import { UserWarning } from './UserWarning';
+import React, { useCallback, useEffect, useState } from 'react';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+import { Footer } from './components/Footer/Footer';
+import { Header } from './components/Header/Header';
+import { ErrorField } from './components/ErrorField/ErrorField';
+import { FilterBy, ERROR } from './types/enums';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterQwery, setFilterQwery] = useState(FilterBy.All);
+  const [errorMessage, setErrorMessage] = useState<ERROR | string>(
+    ERROR.default,
+  );
+  const [todosLoading, setTodosLoading] = useState<number[]>([]);
+
+  // if (!USER_ID) {
+  //   return <UserWarning />;
+  // }
+
+  const loadTodos = useCallback(async () => {
+    try {
+      setErrorMessage(ERROR.default);
+      const data = await getTodos();
+
+      setTodos(data);
+    } catch (err) {
+      setErrorMessage(ERROR.todos);
+    } finally {
+    }
+  }, []);
+
+  useEffect(() => {
+    setTodosLoading([0]);
+    loadTodos();
+    setTodosLoading([]);
+  }, []);
+
+  const adIdToLoadingList = useCallback((id: number) => {
+    setTodosLoading(prev => {
+      if (prev.length > 0) {
+        return [...prev, id];
+      }
+
+      return [id];
+    });
```

Comment:
it can be simplified, spreading empty array wont cause an error

```suggestion
    setTodosLoading(prev => [...prev, id]);
```